src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,257 @@
 package infrastructure.content;
 
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ModuleID;
+import java.util.HashMap;
+import networking.CommunicatorFactory;
+import networking.ICommunicator;
+import networking.INotificationHandler;
+import org.json.JSONObject;
+
+/**
+ * This class ContentCommunicator implements the IContentCommunicator interface and all the 
+ * methods inside it. The ideal purpose of this class is to initialise any new client, send 
+ * message from UI to networking module, handle client exit with respect to content module
+ * and handle the subscribers of content module.
+ * 
+ * @author Badal Kumar (111701008)
+ */
 public class ContentCommunicator implements IContentCommunicator{
-    
+	/**
+	 * This variable will store the username/userid of client as a String.
+	 */
+	private static String userName;
+	
+	/**
+	 * This variable will store the image of client in String format.
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String.
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private static HashMap<String, String> imageMap = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module,
+	 * identifier is of type String while handler is of type IContentNotificationHandler.
+	 */
+	private static HashMap<String, IContentNotificationHandler> handlerMap = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * An instance of ServerPort class is fetched and stored in the variable serverPort.
+	 */

```

Comment:
The two hash maps are made static so that our handler given to networking (NetworkMessageHandler.java) can access these variables which are needed in the receiving flow

Reply:
What has `static` anything to do with access? For access controls we have access specifiers. `static` tells you that you are going to share this member or method over all instances of that class.

Reply:
The NetworkMessageHandler(NMH) needs access to the hashmaps in ContentCommunicator(CC) in order to send data to UI. The initial idea was to have an object of ContentCommunicator inside it and access the variables that way(The ContentFactory ensures single instance of CC). Then we didn't know where NMH will be instantiated and we decided to instantiate it in CC's constructor. We didn't understand how having an NMH object inside an CC object which in turn has a CC object will work so we made the shared variables and methods as static for NMH to use without a CC object instance